#!/usr/bin/env python
PACKAGE = "lsd_slam_to_pcl"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_near_support", int_t, 0, "only plot points that have #min_near_support similar neighbours (higher values remove outliers)", 7, 0, 9)
gen.add("sparsify_factor", int_t, 0, "only plot one out of #sparsify_factor points, selected at random. Use this to significantly speed up rendering for large maps.", 1, 1, 100)
gen.add("depth_var_thresh_abs", double_t, 0, "log10 of threshold on point's variance, in absolute scale.", -1, -10, 1)
gen.add("depth_var_thresh_scaled", double_t, 0, "log10 of threshold on point's variance, in the respective keyframe's scale. ", -3, -10, 1)

exit(gen.generate(PACKAGE, "lsd_slam_to_pcl", "Params"))